### PR TITLE
feat: create unversioned API version for EraserConfig

### PR DESCRIPTION
**Pull request by pmengelbert**
**What this PR does / why we need it**:

The `conversion-gen` tool was not picking up the `EraserConfig` objects, so we were not able to get conversions between them.

This pr:
- Adds `v1alpha2.EraserConfig` and `unversioned.EraserConfig` in addition to the `v1alpha1` version.
- In `v1alpha2` and `unversioned`, The `EraserConfig.Components.Eraser` field has been rename to `EraserConfig.Components.Remover` to accommodate planned changes.
- Removes `ControllerManagerConfigurationSpec` as a field on the `EraserConfig` types.
- Adds auto-generated conversions between `v1alpha1` or `v1alpha2` and `unversioned`.
- Adds a manual conversion function for `v1alpha1 -> unversioned`, because the field name changed from `Eraser` to `Remover`. These functions are necessary and fix the autogenerated conversions.
- Introduces manual specification of `apiVersion` and `kind` in the eraser config schema.
- Introduces manual deserialization of the config format (removing `options.AndFrom...`)
- Fixes the broken `build/version.sh` script, which was setting `version.buildVersion` instead of `version.BuildVersion`.
- Fixes the overlaying of user configuration on top of the defaults. It was previously broken.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: main.go
Comment (pmengelbert) [inline main.go]: @sozercan This seems like a hack around what's supposed to happen. On the plus side we get the auto-generated conversion functions, but I would assume that there's a better way to do this.

At the very worst we can remove the `ControllerManagerConfigurationSpec` as a member of EraserConfig and instead require an `apiVersion` that we manage ourselves. It would have to be added to the helm chart.

File: main.go
Comment (pmengelbert) [inline main.go]: proposed: use this little struct to unmarshal just the APIVersion. Then Convert whatever is received to `unversioned` using the appropriate auto-generated function.

Unmarshaling this way is needed because of the way `TypeMeta`'s `UnmarshaJSON` method is defined.

File: main.go
Comment (sozercan) [inline main.go]: typo here

File: api/unversioned/config/config.go
Comment (sozercan) [inline api/unversioned/config/config.go]: is the comment intentional?

File: api/v1alpha1/eraserconfig_types.go
Comment (sozercan) [inline api/v1alpha1/eraserconfig_types.go]: looks like we need to revert this change?

File: main.go
Comment (sozercan) [inline main.go]:
```suggestion
		setupLog.Error(err, "cannot unmarshal yaml", "bytes", string(fileBytes), "apiVersion", av)
```

File: go.mod
Comment (sozercan) [inline go.mod]: can we merge with other requires?

File: api/v1alpha1/eraserconfig_types.go
Comment (sozercan) [inline api/v1alpha1/eraserconfig_types.go:158]: can we add a brief comment on why we adding manual conversion

File: Makefile
Comment (ashnamehrotra) [inline Makefile:25]: this doesn't seem to compile, I think we need to escape the #: 's\#,$$\#\#' to 's\#,$$\#\#'

Comment (sozercan) [reply Makefile:25]: i am okay to just adding paths manually for now since this is not going to get updated frequently

Comment (pmengelbert) [reply Makefile:25]: I listed them manually for now

File: api/unversioned/config/config.go
Comment (pmengelbert) [reply api/unversioned/config/config.go]: No. This was from when my brain go confused about overlaying on the defaults. It's fine to just have `version.BuildVersion` there.

Comment (pmengelbert) [reply api/unversioned/config/config.go]: Fixed

File: api/v1alpha1/eraserconfig_types.go
Comment (pmengelbert) [reply api/v1alpha1/eraserconfig_types.go]: Done

File: api/v1alpha1/eraserconfig_types.go
Comment (pmengelbert) [reply api/v1alpha1/eraserconfig_types.go:158]: done

File: go.mod
Comment (pmengelbert) [reply go.mod]: done

File: main.go
Comment (pmengelbert) [reply main.go]: done

File: api/unversioned/config/config.go
Comment (ashnamehrotra) [inline api/unversioned/config/config.go]: changed repo to "remover"

**Review by sozercan — APPROVED** [review]
LGTM if this works with @ashnamehrotra's PR

File: api/unversioned/config/config.go
Comment (pmengelbert) [reply api/unversioned/config/config.go]: done

**Comment by pmengelbert** [comment]
@ashnamehrotra This should work as a base for your #675 . LMK if there's anything else that should be in here before I merge

**Comment by ashnamehrotra** [comment]
(resolved)

**Pull request merged**
